src/interfaces/buttonGroup.d.ts

```diff
@@ -0,0 +1,13 @@
+// <reference types="react" />
+import { ButtonSize } from "./buttons";
+
+export interface IButtonGroupProps {
```

Comment:
How about extending `React.HTMLAttributes<HTMLDivElement>` here? You are using `restProps` in the render method of the component so it will do no harm to allow data attributes and so on.

Reply:
Good point will do 🙂